client/components/domains/map-domain-step/index.jsx

```diff
@@ -139,33 +141,33 @@ const MapDomainStep = React.createClass( {
 					onButtonClick={ this.registerSuggestedDomain } />
 			</div>
 		);
-	},
+	}
 
-	registerSuggestedDomain: function( event ) {
+	registerSuggestedDomain = ( event ) => {
 		event.preventDefault();
```

Comment:
This triggers an error when on the domain mapping step and you try to search for an available domain, it suggests to register it. When you click on that registration, you get the following error:
```
Uncaught TypeError: event.preventDefault is not a function
    at Object.MapDomainStep._this.registerSuggestedDomain [as onButtonClick] (index.jsx:85)
    at DomainRegistrationSuggestion._this.onButtonClick (index.jsx:74)
    at Object.ReactErrorUtils.invokeGuardedCallback (ReactErrorUtils.js:70)
    at executeDispatch (EventPluginUtils.js:85)
    at Object.executeDispatchesInOrder (EventPluginUtils.js:108)
    at executeDispatchesAndRelease (EventPluginHub.js:43)
    at executeDispatchesAndReleaseTopLevel (EventPluginHub.js:54)
```

Basically, `event` is not a click event in that instance, but a domain suggestion. Probably a result of all that bind shenanigans?